src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,163 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node;
+        if (size == 0) {
+            node = new Node<T>(null, value, null);
+            tail = node;
+            head = node;
+        } else {
+            node = new Node<T>(tail, value, null);
+            tail.next = node;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexAddByIndex(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            head = new Node<>(null, value, getNodeByIndex(index));
+            size++;
+            return;
+        }
+        Node<T> newNode = new Node<>(getNodeByIndex(index).prev, value, getNodeByIndex(index));
+        getNodeByIndex(index).prev.next = newNode;
+        getNodeByIndex(index).prev = newNode;
+        size++;
```

Comment:
You don't need to call the getNodeByIndex method so often. Just write the result to a variable and use it